codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoderTest.java

```diff
@@ -617,26 +622,61 @@ public Void answer(InvocationOnMock invocationOnMock) throws Throwable {
                 flowControlled.writeComplete();
                 return null;
             }
-        }).when(remoteFlow).addFlowControlled(eq(stream), payloadCaptor.capture());
+        }).when(remoteFlow).addFlowControlled(any(Http2Stream.class), payloadCaptor.capture());
     }
 
-    private void mockFutureAddListener(boolean success) {
-        when(future.isSuccess()).thenReturn(success);
-        if (!success) {
-            when(future.cause()).thenReturn(new Exception("Fake Exception"));
-        }
-        doAnswer(new Answer<Void>() {
+    private Http2Stream createStream(int streamId, boolean halfClosed) throws Http2Exception {
+        return connection.local().createStream(streamId, halfClosed);
+    }
+
+    private Http2Stream reservePushStream(int pushStreamId, Http2Stream parent) throws Http2Exception {
+        return connection.local().reservePushStream(pushStreamId, parent);
+    }
+
+    private Http2Stream stream(int streamId) {
+        return connection.stream(streamId);
+    }
+
+    private void goAwayReceived(int lastStreamId) {
+        connection.goAwayReceived(lastStreamId, 0, Unpooled.EMPTY_BUFFER);
+    }
+
+    private void goAwaySent(int lastStreamId) {
+        connection.goAwaySent(lastStreamId, 0, EMPTY_BUFFER);
+    }
+
+    private ChannelPromise newPromise() {
+        return new DefaultChannelPromise(channel, ImmediateEventExecutor.INSTANCE);
+    }
+
+    private ChannelPromise newVoidPromise() {
+        return new DefaultChannelPromise(channel, ImmediateEventExecutor.INSTANCE) {
             @Override
-            public Void answer(InvocationOnMock invocation) throws Throwable {
-                ChannelFutureListener listener = (ChannelFutureListener) invocation.getArguments()[0];
-                listener.operationComplete(future);
-                return null;
+            public ChannelPromise addListener(
+                    GenericFutureListener<? extends Future<? super Void>> listener) {
+                throw new AssertionFailedError();
```

Comment:
Actually let's keep it as-is ... this is only 1 line of code. If I used `Assert.fail` I would need 2 lines, since the method would have to return something.